### PR TITLE
acrn-config: Support pci vuart configuration and generate pci vuart dev information

### PR DESCRIPTION
**Pull request by yuchuyang**
- assign vbar and vbdf to pci vuart
- insert vuart pci dev to pci_dev.c
- update vm_configurations.c and misc_cfg.h accordinly
- sanity check of configuration error

Tracked-On: #5425
Signed-off-by: Yang, Yu-chu <yu-chu.yang@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Review by terryzouhao — APPROVED** [review]

File: misc/acrn-config/board_config/vbar_base_h.py [hunk 1]
```diff
@@ -128,6 +138,66 @@ def find_next_bar(bar_val, bar_list):
...
+                             PCI_VUART_VBAR0_SIZE + PCI_VUART_VBAR0_ALIGNMENT)
+                free_bar0_start_addr = common.round_up(free_bar0.start, PCI_VUART_VBAR0_ALIGNMENT)
+                free_bar0_end_addr = free_bar0_start_addr + PCI_VUART_VBAR0_SIZE - 1
+                free_bar0 = MmioWindow(free_bar0_start_addr, free_bar0_end_addr)
+                mmiolist_per_vm[vm_id].append(free_bar0)
+                mmiolist_per_vm[vm_id].sort()
+                free_bar1 = get_free_mmio(sos_mmio_range, mmiolist_per_vm[vm_id], \
+                             PCI_VUART_VBAR1_SIZE + PCI_VUART_VBAR1_ALIGNMENT)
+                free_bar1_start_addr = common.round_up(free_bar1.start, PCI_VUART_VBAR1_ALIGNMENT)
+                free_bar1_end_addr = free_bar0_start_addr + PCI_VUART_VBAR1_SIZE - 1
```
Comment (conghuic23) [inline misc/acrn-config/board_config/vbar_base_h.py:176]: I guess this line should be:
free_bar1_end_addr = free_bar1_start_addr + PCI_VUART_VBAR1_SIZE - 1

File: misc/acrn-config/board_config/vbar_base_h.py [hunk 2]
```diff
@@ -128,6 +138,66 @@ def find_next_bar(bar_val, bar_list):
...
+                                mmiolist_per_vm[vm_id], PCI_VUART_VBAR0_SIZE + PCI_VUART_VBAR0_ALIGNMENT)
+                free_bar0_start_addr = common.round_up(free_bar0.start, PCI_VUART_VBAR0_ALIGNMENT)
+                free_bar0_end_addr = free_bar0_start_addr + PCI_VUART_VBAR0_SIZE - 1
+                free_bar0 = MmioWindow(free_bar0_start_addr, free_bar0_end_addr)
+                mmiolist_per_vm[vm_id].append(free_bar0)
+                mmiolist_per_vm[vm_id].sort()
+                free_bar1 = get_free_mmio([MmioWindow(start=common.SIZE_2G, end=common.SIZE_4G-1)], \
+                                mmiolist_per_vm[vm_id], PCI_VUART_VBAR1_SIZE + PCI_VUART_VBAR1_ALIGNMENT)
+                free_bar1_start_addr = common.round_up(free_bar1.start, PCI_VUART_VBAR1_ALIGNMENT)
+                free_bar1_end_addr = free_bar0_start_addr + PCI_VUART_VBAR1_SIZE - 1
```
Comment (conghuic23) [inline misc/acrn-config/board_config/vbar_base_h.py:191]: Ditto

**Review by conghuic23 — CHANGES_REQUESTED** [review]

**Comment by wenlingz** [comment]
@yuchuyang  please check conghui's comments

File: misc/acrn-config/board_config/vbar_base_h.py (hunk 1 above)
Comment (yuchuyang) [reply misc/acrn-config/board_config/vbar_base_h.py]: fixed

File: misc/acrn-config/board_config/vbar_base_h.py (hunk 2 above)
Comment (yuchuyang) [reply misc/acrn-config/board_config/vbar_base_h.py]: fixed

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Pull request merged**